actors/miner/src/lib.rs

```diff
@@ -5008,16 +5012,31 @@ fn batch_activate_deals_and_claim_allocations(
     for (activation_info, activate_res) in
         successful_activation_infos.iter().zip(&batch_activation_res.activations)
     {
-        let sector_claims = activate_res
-            .verified_infos
-            .iter()
-            .map(|info| ext::verifreg::AllocationClaim {
-                client: info.client,
-                allocation_id: info.allocation_id,
-                data: info.data,
-                size: info.size,
-            })
-            .collect();
+        let mut total_verified_space = BigInt::zero();
+        let mut sector_claims = Vec::new();
+        for verified in &activate_res.verified_infos {
+            total_verified_space += verified.size.0;
+            sector_claims.push(ext::verifreg::AllocationClaim {
+                client: verified.client,
+                allocation_id: verified.allocation_id,
+                data: verified.data,
+                size: verified.size,
+            });
+        }
+
+        // Verify the deals for each sector fit in the sector size.
+        // This check is redundant if the sector's unsealed CID (CommD) is verified as
+        // compatible with the deal pieces.
```

Comment:
I don't understand why we are doing this here.

Here and when using this for the new ProveCommit that calls SectorContentChanged we must always be ensuring that the sector's unsealed CID is compatible with the deal pieces.  We won't be able to avoid doing an inclusion proof of all Piece Cids inside the CommD without breaking critical guarantees of FIL+ / QAP.  Because of this this code seems like it must be redundant.

Reply:
I put this here because I removed an equivalent check from BatchActivateDeals, which no longer calls `validate_deals_for_sector` but validates the deals independently. 

I agree that it's probably redundant, but it's quite hard to think through the dataflow of the different cases to be sure. For initial onboarding, with deals specified at precommit, I think we're ok because the CommD check is made at pre-commit (via the market actor :-(). If deals are deferred until the new ProveCommit, then that method will need do an equivalent check in the miner, i.e. the right way.

For old replica update, there's currently a mess where it calls VerifyDealsForActivation, which _does_ call `validate_deals_for_sector`, after actually activating them, and so the market actor does the check. We should change this #1308 to make only one call and move the commd equivalence check into the miner. New replica update should do it the right way.

Sooo, yes I think it's redudant, this was defensive because I removed it from the market actor. Given assumption that we're going to check CommD == Deals _somewhere_, I'm happy to remove this. But that's the context for your consideration.

Reply:
Reverted